pkg/cmd/taskrun/cancel_test.go

```diff
@@ -62,17 +62,28 @@ func TestTaskRunCancel(t *testing.T) {
 	}
 
 	trs2 := []*v1alpha1.TaskRun{
-		tb.TaskRun("failure-taskrun-1",
-			tb.TaskRunNamespace("ns"),
-			tb.TaskRunLabel("tekton.dev/task", "failure-task"),
-			tb.TaskRunSpec(tb.TaskRunTaskRef("failure-task")),
-			tb.TaskRunStatus(
-				tb.StatusCondition(apis.Condition{
-					Status: corev1.ConditionTrue,
-					Reason: v1beta1.TaskRunReasonFailed.String(),
-				}),
-			),
-		),
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "failure-taskrun-1",
+				Namespace: "ns",
+				Labels:    map[string]string{"tekton.dev/task": "failure-task"},
+			},
+			Spec: v1alpha1.TaskRunSpec{
+				TaskRef: &v1alpha1.TaskRef{
+					Name: "failure-task",
+				},
+			},
+			Status: v1alpha1.TaskRunStatus{
+				Status: duckv1beta1.Status{
+					Conditions: duckv1beta1.Conditions{
+						{
+							Status: corev1.ConditionUnknown,
```

Comment:
This should be changed to Unknown since it needs to make it through the validation of whether or not the TaskRun has finished or not.